From bd7e3ced7bf89917541eab7b74e1aa844f049cc5 Mon Sep 17 00:00:00 2001
From: Leon Romanovsky <leonro@mellanox.com>
Date: Wed, 2 Oct 2019 15:51:53 +0300
Subject: [PATCH] smatch: Fix compilation error

make: *** No rule to make target 'check_list_local.h', needed by
'smatch.o'.  Stop.
make: *** Waiting for unfinished jobs....
  CC      sparse-llvm.o

Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
---
 Makefile      | 2 +-
 check_list.h  | 2 +-
 pre-process.c | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 7fc69162..57fce22e 100644
--- a/Makefile
+++ b/Makefile
@@ -257,7 +257,7 @@ SMATCH_LDFLAGS := -lsqlite3  -lssl -lcrypto
 smatch: smatch.o $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS)
 	$(QUIET_LINK)$(LD) -o $@ $< $(SMATCH_FILES) $(SMATCH_CHECKS) $(LIBS) $(SMATCH_LDFLAGS)
 
-smatch.o: smatch.c $(LIB_H) smatch.h check_list.h check_list_local.h
+smatch.o: smatch.c $(LIB_H) smatch.h check_list.h
 	$(CC) $(CFLAGS) -c smatch.c -DSMATCHDATADIR='"$(SMATCHDATADIR)"'
 
 $(SMATCH_CHECKS): smatch.h smatch_slist.h smatch_extra.h avl.h
diff --git a/check_list.h b/check_list.h
index e085cea4..84a33fb4 100644
--- a/check_list.h
+++ b/check_list.h
@@ -195,7 +195,7 @@ CK(check_implicit_dependencies)
 CK(check_wine_filehandles)
 CK(check_wine_WtoA)
 
-#include "check_list_local.h"
+//#include "check_list_local.h"
 
 CK(register_scope)
 CK(register_stored_conditions)
diff --git a/pre-process.c b/pre-process.c
index c35cbcfb..08601754 100644
--- a/pre-process.c
+++ b/pre-process.c
@@ -606,7 +606,7 @@ static struct token *dup_token(struct token *token, struct position *streampos)
 	alloc->pos.whitespace = token->pos.whitespace;
 	alloc->number = token->number;
 	alloc->pos.noexpand = token->pos.noexpand;
-	return alloc;	
+	return alloc;
 }
 
 static struct token **copy(struct token **where, struct token *list, int *count)
@@ -1383,7 +1383,7 @@ static struct token *handle_hashhash(struct token *token, struct token *arglist)
 	struct token *last = token;
 	struct token *concat;
 	int state = match_op(token, ',');
-	
+
 	try_arg(token, TOKEN_QUOTED_ARGUMENT, arglist);
 
 	while (1) {
-- 
2.20.1

